enhancements/ingress/add-headers-to-haproxyconfig-via-ingress-controller-or-route-api.md

```diff
@@ -0,0 +1,295 @@
+---
+title: add-headers-to-haproxyconfig-via-ingress-controller-or-route-api
+authors:
+  - "@miheer"
+reviewers:
+  - "@frobware"
+  - "@gcs278"
+  - "@Miciah"
+approvers:
+  - "@frobware"
+  - "@gcs278"
+  - "@Miciah"
+api-approvers:
+  - "@frobware"
+  - "@gcs278"
+  - "@Miciah"
+creation-date: 2020-06-10
+last-updated: 2020-06-10
+tracking-link:
+  - "https://issues.redhat.com/browse/NE-982"
+status: implemented
+see-also:
+- ""
+replaces:
+superseded-by:
+---
+
+# Set HTTP Headers to haproxy.config either via Ingress Contoller or Route API
+
+## Summary
+
+This enhancement extends the IngressController logging API and router API to allow the user to set 
+capture of HTTP headers in the haproxy.config file. Using the new API, users may specify a list of 
+HTTP headers either via IngressController API or Route API that should be set in the haproxy.config file.
+
+## Motivation
+
+Certain headers may need to be set for compliance reasons or other reasons. In OpenShift 3, cluster administrators could
+use a custom HAProxy configuration template to set headers, but using a custom template is not permitted in OpenShift 4.
+
+### User Stories
+
+As a cluster administrator, I need to configure an IngressController to set certain HTTP headers,
+in order to satisfy compliance requirements.
+To satisfy this use-case, the cluster administrator can either set the IngressController's or Router's spec.httpSetHeaders
+field to set the required headers.
+
+### Goals
+
+1. Enable the cluster administrator to specify HTTP headers in the IngressController CR Spec that should get set in the global section
+   of the haproxy.config.
+2. Enable the developer or user to specify HTTP headers in the Route Spec that should get set in the backend section `default_backend openshift_default`.
+   of the haproxy.config.
+
+### Non-Goals
+
+1. Admins or users are responsible for providing proper header values as inputs.
```

Comment:
BTW, reading the RFE, I didn't quite get whether the headers should be set for the responses or requests. I tend to read it for like for the responses.

Reply:
@alebedev87, I believe you are reading the RFE correctly.

Reply:
Removed the question as it was appearing outside this thread, my bad, I didn't know it would disappear here too. The question was about the request and response headers, I thought that "by RFE" we only need to implement the response headers.